src/matching/Matching_generic.ml

```diff
@@ -853,3 +883,12 @@ let m_other_xxx a b =
   match (a, b) with
   | a, b when a =*= b -> return ()
   | _ -> fail ()
+
+let m_doc_string_metavar_or_string a b =
+  let (a_str, _) = a in
+  if Mvar.is_metavar_name a_str
+  then
+    let (b_str, b_tok) = b in
+    let (trimmed_str, trimmed_tok) = adjust_info_remove_doc_string_delimiters (b_str, b_tok) in
+    envf a (MV.Text (trimmed_str, trimmed_tok, b_tok))
+  else fail ()
```

Comment:
Why `else fail ()` ? 🤔 What if the pattern is exactly the same as the code, I think we should match.